index.js

```diff
@@ -189,27 +241,20 @@ internals.write = function(output) {
  * @return {Boolean} true if enabled
  */
 internals.isEnabled = function(namespace, level) {
-    if (_.indexOf(internals.levels, level) < _.indexOf(internals.levels, internals.level)) {
-        return false
-    }
-
-    let i = 0
-    while (i < internals.disabledNamespaces.length) {
-        if (internals.disabledNamespaces[i].test(namespace)) {
-            return false
-        }
-        i += 1
-    }
-
-    let j = 0
-    while (j < internals.enabledNamespaces.length) {
-        if (internals.enabledNamespaces[j].test(namespace)) {
-            return true
+    let nsLevel = internals.level
+    let nsMatch = false
+
+    _.forEachRight(internals.namespaces, (ns) => {
```

Comment:
You could use this approach to avoid executing all regex:

```javascript
const namespaceConfig  = _.findLast(internals.namespaces, ({ regex }) => regex.test(namespace))
```